### PR TITLE
Monorepo hacking

### PR DESCRIPTION
**Pull request by mythmon**
This is kind of ugly, as it is bringing a few other repo's entire history into this repo, but I think it is a good idea to maintain history.

* [x] Merge everything in to one repo
* [x] Set up CI framework
* [x] Set up linting
* [x] Move docs
* [x] Write mono-readme

### CODE REVIEW

File: compose/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+This is a repo to hold all the bits to get a prod-like Normandy running using
+[Docker Compose][].
+
+1. Install [Docker][] and Docker Compose, and start the Docker daemon (possibly via [Docker Machine][]).
+2. Clone this repo, and `cd` into it.
```
Comment (andymikulski) [inline compose/README.md]: I found I had to cd into the `compose` directory to get the following steps to work. Is that a missing instruction or should all the `docker-compse` stuff be available from the root dir?

File: compose/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+3. `./bin/genkeys.sh` to create keys for HTTPS.
+4. `docker-compose up -d` to download Docker images, and run them in the background.
+5. `docker-compose run normandy ./manage.py migrate` to create the needed database tables.
+6. `docker-compose run normandy ./manage.py createsuperuser` to create a user you can log in with.
+6. `docker-compose run normandy ./manage.py update_actions` to load the action code into the database.
```
Comment (andymikulski) [inline compose/README.md]: Two `6.`s!

File: compose/README.md (hunk 1 above)
Comment (mythmon) [inline compose/README.md]: I haven't updated these instructions to account for the monorepo yet. I'll probably do that as a unified docs pass.

**Comment by mythmon** [comment]
Ok, I think this is ready to go, though there are a few things I'm going to file follow ups for.

"Normandy" is now the collection of projects. "recipe-server" is what normandy used to be, "recipe-client-addon" is the system add-on, etc.

Most of this is just moved from the other repos, and you can safely ignore almost all the commits. The commits to look at start with "Monorepo CI framework". Things to look at especially are the contents of `bin/ci` at the top level, and in every project's directory, as well as READMEs in the same places.

Remaining issues:

* CI artifacts aren't properly gathered by CircleCI yet, so we don't have complete failure details.
* Docs need rewritten. Only recipe-server had any docs to speak of, but they need to be reorganized and re-worded. I'd also like to put set-up and development docs for every project on readthedocs, instead of in READMEs.
* The badges in the README are probably broken, and need looking into, but I think that must wait until this merges, so the tools can pick up on the organization.
* Probably a million other things that will pop up as we work through this.

**Comment by mythmon** [comment]
@chartjes I didn't change the actual tests here, but the way they get run in CI is different. It is probably worth looking over, to make sure you can work with this new format.

**Comment by mythmon** [comment]
To clarify, CI failures will be reported as failures, and all the details are available. There is a regression in niceness in that the failure details are buried in the logs, instead of being surfaced by CircleCI's interface.

**Comment by chartjes** [comment]
@mythmon Understood

File: lints/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Lints
+
+This is a unified linting tool for all Normandy project.
```
Comment (Osmose) [inline lints/README.md]: projects?

File: lints/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# Usage
+
+The lints are run in a Docker container which sets up all tools needed. To
+provide the code to lint, mount it as a Docker volume at ``/app/code`. The
```
Comment (Osmose) [inline lints/README.md]: Missed a backtick.

File: recipe-server/README.md [hunk 5]
```diff
@@ -1,19 +1,10 @@
...
+# Recipe Server
 
-Normandy is the server-side implementation of the [Recipe Server][]. It serves
-bundles of JavaScript to various clients (Firefox browsers) based on certain
-rules.
-
-[![CircleCI](https://img.shields.io/circleci/project/mozilla/normandy/master.svg?maxAge=2592000&label=CI)](https://circleci.com/gh/mozilla/normandy/tree/master)
-[![Requires.io](https://img.shields.io/requires/github/mozilla/normandy.svg?maxAge=2592000&label=PyPI)](https://requires.io/github/mozilla/normandy/requirements/?branch=master)
-[![David](https://img.shields.io/david/mozilla/normandy.svg?maxAge=2592000&label=NPM)](https://david-dm.org/mozilla/normandy)
+The Recipce Server serves bundles of JavaScript to various clients (Firefox
```
Comment (Osmose) [inline recipe-server/README.md]: Spelled recipe wrong.

**Pull request merged**